bluesky/callbacks/zmqsub.py

```diff
@@ -0,0 +1,102 @@
+import time
+from ..utils import expiring_function
+from ..run_engine import Dispatcher, DocumentNames
+
+
+_loop = zmq.asyncio.ZMQEventLoop()
+asyncio.set_event_loop(_loop)
+
+
+class RemoteDispatcher(Dispatcher):
+    def __init__(self, host, port, *, filter_hostname=None, filter_pid=None,
+                 filter_run_engine_id=None, event_timeout=None):
+        """
+        Dispatch documents received over a socket.
+
+        Parameters
+        ----------
+        host : string
+            name of host running forwarder_device
+            (See bluesky/examples/forwarder_device.py.)
+        port : int, optional
+            default 5560
+        filter_hostname : string, optional
+            only process documents from a host with this name
+        filter_pid : int, optional
+            only process documents from a process with this pid
+        filter_run_engine_id : int, optional
+            only process documents from a RunEngine with this Python id
+            (memory address)
+        event_timeout : float
+            expiring time for skipping an event
+
+        Example
+        -------
+        Create a LivePlot and feed it documents published to a 0MQ forwarder
+        device at running on localhost at port 5560.
+
+        >>> from bluesky.qt_kicker import install_qt_kicker
+        >>> install_qt_kicker()
+        >>> dispatcher = Dispatcher('localhost', 5568)
+        >>> dispatcher.subscribe(LivePlot('y', 'x'))
+        >>> dispatcher.start()
+        """
+        self._host = host
+        self._port = int(port)
+        self._context = zmq.asyncio.Context()
+        self._socket = self._context.socket(zmq.SUB)
+        url = "tcp://%s:%d" % (self.host, self.port)
+        self._socket.connect(url)
+        self._socket.setsockopt_string(zmq.SUBSCRIBE, "")
+        self.event_timeout = event_timeout
+
+        def is_our_message(hostname, pid, RE_id):
+            # Close over filters and decide if this message applies to this
+            # RemoteDispatcher.
+            return ((filter_hostname is None
+                     or filter_hostname == hostname)
+                    and (filter_pid is None
+                         or filter_pid == pid)
+                    and (filter_run_engine_id is None
+                         or filter_run_engine_id == RE_id))
+        self._is_our_message = is_our_message
+
+        super().__init__()
+
+    @property
+    def host(self):
+        return self._host
+
+    @property
+    def port(self):
+        return self._port
+
+    @asyncio.coroutine
+    def _poll(self):
+        message = yield from self._socket.recv()
+        hostname, pid, RE_id, name, doc = message.decode().split(' ', 4)
+        doc = ast.literal_eval(doc)
+        self._loop.create_task(self._poll())
+        start_time = time.time()  # for timeout, if used below
+        yield from asyncio.sleep(0)  # Give scheduler second chance to switch.
+        if self._is_our_message(hostname, pid, RE_id):
```

Comment:
As I mentioned in person, I think it will be important to come up with a standardized way to allow access to bundled messages (or the whole queue). 

For example, if there are 10 points to be added to a 2D `imshow`/`LiveImage`, the plot does not have to be redrawn for each point, but only once with the full set. It would not be appropriate to drop those points nor send them out of order just because the routine takes too long to redraw.